### PR TITLE
Use dynamic suffix for service account secrets and allow updates of service account annotations

### PR DESCRIPTION
**Pull request by jswoods**
This pr does two things:

1) Move to dynamic suffixes for the secrets created for the service accounts. The reasoning here is if these secrets change, the pods will still attempt to use the old secrets. We rely on the dynamic suffix to trigger pod deletions to pick up the new secret name. This is not directly related to the other change in this PR, but something I noticed during my testing and something we should change now as it will be more painful later on. If you look at secrets that are type of `kubernetes.io/service-account-token` they are always named with a hash suffix. 

2) Allow updates of the service account annotations. 

Another option I explored for this was instead of doing an update on the service account resource when we change annotations, we move the service account to a dynamic/hash suffix and then do a delete/recreate similar to how we handle pods. In this case, the service account is recreated and so is the secret associated with it, however this would also trigger a recreation the pods. In the case of adding the annotation `eks.amazonaws.com/role-arn` (which is what we use this for), we do not require that we recreate the service account or secret, hence why we can just do the update. But we still need to manually delete the pods. Perhaps there may be other cases where we do need to recreate the service account and/or secret when changing a service account. In this case, because of the work of #1 in the PR, we have the option of recreating the service account and then the secret and pods will also be recreated.

So my questions are: should we move the service account to use a dynamic suffix like the secrets? Or should we leave the service account statically named but perhaps we do something to the pod to trigger recreation when the service account is changed, like delete/recreate the service account secret? Or do we leave the update logic and rely on manually deleting the pods?

### CODE REVIEW

**Comment by jswoods** [comment]
Goal if this is to fix https://github.com/humio/humio-operator/issues/116.

**Comment by jswoods** [comment]
Let's continue with this and once we have a fix for https://github.com/humio/humio-operator/issues/122 (which should be soon, as I'm working on another PR for restart functionality) then I will make the updates so that we roll the humio pods after changing the service account annotations (added a TODO for this).

**Comment by jswoods** [comment]
I have rebased and refactored a bit after pulling in the restart functionality. This should be ready to go now.

File: deploy/crds/core.humio.com_humioclusters_crd.yaml [hunk 1]
```diff
@@ -2,12 +2,6 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: humioclusters.core.humio.com
-  labels:
-    app: 'humio-operator'
-    app.kubernetes.io/name: 'humio-operator'
-    app.kubernetes.io/instance: 'humio-operator'
-    app.kubernetes.io/managed-by: 'Helm'
-    helm.sh/chart: 'humio-operator-0.0.6'
```
Comment (SaaldjorMike) [inline deploy/crds/core.humio.com_humioclusters_crd.yaml:10]: Hmm. Those labels should get added automatically when running "make crds". Did you generate the CRD's using that make target or did you run `operator-sdk generate crds` directly?

File: pkg/controller/humiocluster/humiocluster_controller.go [hunk 2]
```diff
@@ -1381,8 +1470,13 @@ func (r *ReconcileHumioCluster) ensurePodsExist(ctx context.Context, hc *corev1a
 		return reconcile.Result{}, err
 	}
 
+	attachments, err := r.newPodAttachments(ctx, hc, foundPodList)
+	if err != nil {
+		r.logger.Errorf("failed to get pod attachments: %s", err)
```
Comment (SaaldjorMike) [inline pkg/controller/humiocluster/humiocluster_controller.go:1475]: @jswoods I think we might be missing a return statement here. I assume we'd want to return the error?

File: deploy/crds/core.humio.com_humioclusters_crd.yaml (hunk 1 above)
Comment (jswoods) [reply deploy/crds/core.humio.com_humioclusters_crd.yaml]: fixed

File: pkg/controller/humiocluster/humiocluster_controller.go (hunk 2 above)
Comment (jswoods) [reply pkg/controller/humiocluster/humiocluster_controller.go]: fixed

File: pkg/controller/humiocluster/humiocluster_controller.go [hunk 3]
```diff
@@ -1350,7 +1434,12 @@ func (r *ReconcileHumioCluster) ensurePodsBootstrapped(ctx context.Context, hc *
 
 	r.logger.Debugf("pod ready count is %d, while desired node count is %d", podsReadyCount, hc.Spec.NodeCount)
 	if podsReadyCount < hc.Spec.NodeCount {
-		err = r.createPod(ctx, hc, foundPodList)
+		attachments, err := r.newPodAttachments(ctx, hc, foundPodList)
+		if err != nil {
+			r.logger.Errorf("failed to get pod attachments: %s", err)
+			return reconcile.Result{}, err
```
Comment (SaaldjorMike) [inline pkg/controller/humiocluster/humiocluster_controller.go:1440]: @jswoods Given that the other `ensurePodsExist` have `return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 5}, err`, perhaps we want to update this one to do the same?

**Pull request merged**